app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java

```diff
@@ -534,9 +535,10 @@ private void VibrateNotifyMakeNoise(Context context, AlertType alert, String bgV
                 .setDeleteIntent(snoozeIntent(context, minsFromStartPlaying));
 
         if (profile != ALERT_PROFILE_VIBRATE_ONLY && profile != ALERT_PROFILE_SILENT) {
-            if (minsFromStartPlaying >= MAX_VIBRATING_MINUTES) {
-                // Before this, we only vibrate...
-                float volumeFrac = (float) (minsFromStartPlaying - MAX_VIBRATING_MINUTES) / (MAX_ASCENDING_MINUTES - MAX_VIBRATING_MINUTES);
+            if (minsFromStartPlaying >= MAX_VIBRATING_MINUTES || (minsFromStartPlaying > Vibrating_Calc && !Pref.getBoolean("delay_ascending_3min", true))) {
+                float volumeFrac = (float) (minsFromStartPlaying - Vibrating_Calc) / (MAX_ASCENDING_MINUTES - Vibrating_Calc);
```

Comment:
This line is an elegant mathematical solution but much more dangerous because it is not easy to understand and it becomes invalid (as well as the comment) as soon as one changes one of the two underlying constants.

Reply:
I don't follow why you have made this comment.
The code as it stands, before this PR, exactly fits your description.

Do you want me to rewrite this code from scratch throwing away what tzachi wrote a decade ago?

I'm not going to do that.
If you want to change the code, you can.  My intent is to remove the delay.